Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -226,8 +227,9 @@ namespace ROS2
 
         for (const auto child : children)
         {
-            auto child_entity = AzToolsFramework::GetEntityById(child);
-
+            AZ::Entity* child_entity;
```

Comment:
minor: initialize to nullptr
```suggestion
            AZ::Entity* child_entity = nullptr;
```